### PR TITLE
Add find feature implementation and design in developer guide

### PR DESCRIPTION
**Pull request by swtan346**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -304,6 +304,43 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+![FindClassDiagram](images/FindClassDiagram.png)
+
+Additionally, it implements the following operations:
+
+* `FindCommandParser#parse()` — Parses user input and creates a `FindCommand` object.
+
+Given below is an example usage scenario and how the find by name or ic mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time.
```
Comment (iamtr) [inline docs/DeveloperGuide.md]: We probably don't need this step

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -304,6 +304,43 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+`NameContainsKeywordsPredicate` and 
+`IcContainsKeywordsPredicate`.
+
+![FindClassDiagram](images/FindClassDiagram.png)
+
+Additionally, it implements the following operations:
+
+* `FindCommandParser#parse()` — Parses user input and creates a `FindCommand` object.
+
+Given below is an example usage scenario and how the find by name or ic mechanism behaves at each step.
```
Comment (iamtr) [inline docs/DeveloperGuide.md]: Nit but would it be better to specify "IC" instead of ic?

**Review by iamtr — COMMENTED** [review]
A few comments regarding the DG.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (swtan346) [reply docs/DeveloperGuide.md]: ok

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (swtan346) [reply docs/DeveloperGuide.md]: sure

**Review by iamtr — APPROVED** [review]
LGTM

**Pull request merged**